content/documentation/pages/5-feature-guides/3-batch/7-java-dsl.md

```diff
@@ -0,0 +1,163 @@
+---
+path: 'feature-guides/batch/java-dsl/'
+title: 'Task Java DSL'
+description: 'Programmatically create tasks using the Java DSL'
+---
+
+# Task & TaskSchedule Java DSL
+
+Instead of using the shell to create and launch tasks, you can use the Java-based DSL provided by the `spring-cloud-dataflow-rest-client` module.
+The Task and TaskScheduler Java DSL provide convenient wrappers around the `DataFlowTemplate` class that enables creating, launching and scheduling tasks programmatically.
+
+To get started, you need to add the following dependency to your project:
+
+```xml
+<dependency>
+	<groupId>org.springframework.cloud</groupId>
+	<artifactId>spring-cloud-dataflow-rest-client</artifactId>
+	<version>%dataflow-version%</version>
+</dependency>
+```
+
+The classes at the heart of the Java DSL are `TaskBuilder`, `Task`, `TaskSchedule`, `TaskScheduleBuilder` and `DataFlowTemplate`.
+The task DSL also utilizes few DataFlowTemplate classes such as `TaskExecutionResource`, `TaskExecutionStatus`, `JobExecutionResource` and `JobInstanceResource`.
+
+The entry point is the `builder` method on `Task` and `TaskSchedule` that takes an instance of a `DataFlowTemplate`.
+
+## Obtain DataFlowTemplate
+
+Both the `Task` and the `TaskSchedule` DSL requires a valid `DataFlowTemplate` instance.
+To create an instance of a `DataFlowTemplate`, you need to provide a `URI` location of the Data Flow Server.
+Spring Boot auto-configuration for `DataFlowTemplate` is also available.
+You can use the properties in [DataFlowClientProperties](https://github.com/spring-cloud/spring-cloud-dataflow/blob/master/spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/config/DataFlowClientProperties.java) to configure the connection to the Data Flow server.
+Generally, you should start with the `spring.cloud.dataflow.client.uri` property.
+
+```Java
+URI dataFlowUri = URI.create("http://localhost:9393");
+DataFlowOperations dataFlowOperations = new DataFlowTemplate(dataFlowUri);
+```
+
+## Task DSL Usage
+
+Consider the following example, which creates a new composed task:
+
+```Java
+dataFlowOperations.appRegistryOperations().importFromResource(
+                     "https://dataflow.spring.io/task-maven-latest", true);
+
+Task task = Task.builder(dataflowOperations)
+              .name("myComposedTask")
+              .definition("a: timestamp && b:timestamp")
+              .description("My Composed Task")
+              .create();
+```
+
+The `TaskBuilder` class is returned from the `Task.builder(dataFlowOperations)` method.
+
+The `create` method returns an instance of a `Task` definition that represents a composite task that has been created but not launched.
+It takes a single string for the task definition (same as in the shell).
+
+<!--NOTE-->
+
+In order to create and launch your tasks, you need to make sure that the corresponding apps have been registered in the Data Flow server first.
```

Comment:
Add this to the end of the sentence.  in the Data Flow server first `as shown [here](%currentPath%/batch-developer-guides/batch/data-flow-spring-batch/#create-task-definition).`